### PR TITLE
EZP-27610 location redirect

### PR DESCRIPTION
**Pull request by mcurran-inviqa**

### CODE REVIEW

File: src/bundle/Controller/TabController.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    ) {
+        $this->router = $router;
+        $this->contentService = $contentService;
+    }
+
+    protected function resetLocation($contentId)
+    {
+        $mainLocationId = $this->contentService->loadContentInfo($contentId)->mainLocationId;
+
+        return new ResetResponse($this->generateTabUrl($contentId, $mainLocationId, 'full'));
```
Comment (maxbaldanza) [inline src/bundle/Controller/TabController.php]: Maybe a default of `full` could be used for the viewType in `generateTabUrl` as that's what the default is of `_ez_content_view`

File: src/bundle/Controller/VersionController.php [hunk 2]
```diff
@@ -7,40 +7,34 @@
...
-use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\Routing\RouterInterface;
 
-class VersionController extends Controller
+class VersionController extends TabController
 {
     public function draftActionsAction(
-        $contentId,
+        Content $content,
```
Comment (maxbaldanza) [inline src/bundle/Controller/VersionController.php:18]: 😮  does this automatically get loaded from the passed value from the route?

File: src/bundle/Controller/VersionController.php [hunk 3]
```diff
@@ -7,40 +7,34 @@
...
         Request $request,
         UiVersionService $versionService,
-        UiFormFactory $formFactory,
-        RouterInterface $router
+        UiFormFactory $formFactory
     ) {
         $draftActionsForm = $formFactory->createVersionsDraftActionForm();
         $draftActionsForm->handleRequest($request);
 
+        $redirectLocationId = $request->query->get('redirectLocationId', $content->contentInfo->mainLocationId);
```
Comment (maxbaldanza) [inline src/bundle/Controller/VersionController.php:26]: Won't this always redirect back to the main content? Is there a reason we're looking for the param from the request?

File: src/bundle/Controller/VersionController.php (hunk 2 above)
Comment (mcurran-inviqa) [reply src/bundle/Controller/VersionController.php:18]: it does

File: src/bundle/Controller/VersionController.php (hunk 3 above)
Comment (mcurran-inviqa) [reply src/bundle/Controller/VersionController.php:26]: it needs to be added to a later PR to fix the urls, they should all pass location but we need a fallback

File: src/bundle/Controller/VersionController.php (hunk 2 above)
Comment (maxbaldanza) [reply src/bundle/Controller/VersionController.php:18]: https://media2.giphy.com/media/ZMyLicvZlX7OM/200.webp#2-grid1

**Pull request closed without merging**